report/src/main/webapp/app/components/compareScreens.directive.js

```diff
@@ -0,0 +1,648 @@
+        imgA: items[0].src,
+        imgB: items[1].src,
+      };
+
+      var canvas = {
+        canvasWrapper: wrapper.querySelector('.canvas-wrapper'),
+        canvasA: document.createElement('canvas'),
+        canvasB: document.createElement('canvas'),
+      };
+
+      var label = {
+        labelA: wrapper.querySelector('.labelA'),
+        labelB: wrapper.querySelector('.labelB'),
+      };
+
+      var ruler = {
+        rulerA: label.labelA.querySelector('.ruler'),
+        rulerB: label.labelB.querySelector('.ruler'),
+      };
+
+      var mask = {
+        maskA: {
+          top: wrapper.querySelector('.maskAtop'),
+          bot: wrapper.querySelector('.maskAbot'),
+        },
+        maskB: {
+          top: wrapper.querySelector('.maskBtop'),
+          bot: wrapper.querySelector('.maskBbot'),
+        },
+        customMask: wrapper.querySelector('.customMasks'),
+        maskSize: null,
+      };
+
+      var context = {
+        contextA: canvas.canvasA.getContext('2d'),
+        contextB: canvas.canvasB.getContext('2d'),
+      };
+
+      var imgSize = {
+        imgSizeA: null,
+        imgSizeB: null,
+      };
+
+      var simple = {
+        simpleA: null,
+        simpleB: null,
+      };
+
+      var cursor = {
+        cursorA: 0,
+        cursorB: 0,
+        cursorBstart: 0,
+      };
+
+      var difs = {
+        main: wrapper.querySelector('.difs'),
+        difsA: [],
+        difsB: [],
+      };
+
+      var invertedGroups = {
+        invertedGroupsA: [],
+        invertedGroupsB: [],
+      };
+
+      var groups = [];
+      var group = [];
+
+      loadImage(img.imgA, imgSize.imgSizeA, canvas.canvasA, context.contextA, simple.simpleA, function (returnedSimple, returnedImgSize) {
+        simple.simpleA = returnedSimple;
+        imgSize.imgSizeA = returnedImgSize;
+      });
+
+      loadImage(img.imgB, imgSize.imgSizeB, canvas.canvasB, context.contextB, simple.simpleB, function (returnedSimple, returnedImgSize) {
+        simple.simpleB = returnedSimple;
+        imgSize.imgSizeB = returnedImgSize;
+        button.classList.remove('button-disabled');
+        tabContent.removeChild(infoMsg);
+        arrangeMasks(mask, difs, canvas, arrangeTimeout, arrangeMaskCallback);
+      });
+
+      function arrangeMaskCallback(maskSize) {
+        mask.maskSize = maskSize;
+        onImagesReady(mask, imgSize, simple, label, ruler, invertedGroups, difs, wrapper);
+        processLines(cursor, simple, group, groups);
+        invertGroups(simple, groups, invertedGroups);
+        drawDifferences(difs, imgSize, invertedGroups, simple, context);
+      }
+
+      canvas.canvasWrapper.appendChild(canvas.canvasA);
+      canvas.canvasWrapper.appendChild(canvas.canvasB);
+    };
+
+    function drawDifferences(difs, imgSize, invertedGroups, simple, context) {
+      var maxHeight = Math.max(imgSize.imgSizeA.height, imgSize.imgSizeB.height);
+      var proc = 100 / maxHeight;
+      var newDiffBLeft = '52%';
```

Comment:
What does this magic number mean?

Reply:
It's the `left` value of second canvas' differences that's being created few lines below. I got no idea how to make it clearer, that second canvas just needs to be pushed by 52% from the left to make both canvases apprear next to each other.